### ISSUE TITLE
 [AC] [Bug] [Core] Data table storage error occurred after updating the latest version

### ISSUE THREAD

**Issue by moncat2005** [trickerer/Trinity-Bots#938]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that my issue may get **closed without notice** if I *intentionally* remove or skip any mandatory\* field

### Current behaviour

After updating to the latest version, some players have reported that their progress in the character game cannot be saved, resulting in lost items.
Upon checking the server errorlog , the following error was found:
2024-11-28 00:23:49 %-5s [sql.sql] [1452] Cannot add or update a child row: a foreign key constraint fails (`acore_characters`.`characters_npcbot_gear_storage`, CONSTRAINT `characters_npcbot_gear_storage_ibfk_1` FOREIGN KEY (`item_guid`) REFERENCES `item_instance` (`guid`) ON DELETE CASCADE ON UPDAT)
2024-11-28 00:23:49 %-5s [sql.sql] Unhandled MySQL errno 1452. Unexpected behaviour possible.

### TrinityCore or AzerothCore

AzerothCore

### Core rev. hash/commit

AzerothCore rev. a88b9b5 2024-11-26 15:39:27 +0800 (npcbots_3.35 branch) 

### Operating system

ubuntu 22.04

**Comment by trickerer** [comment]
No, it's not related. Gear storage capacity doesn't affect the data stored in DB tables.
This error means there is an entry in characters_npcbot_gear_storage with item guid not present in item_instance table which is not possible without manual modification.
The only thing that can be done is to remove invalid entries.
```sql
DELETE FROM characters_npcbot_gear_storage WHERE item_guid NOT IN (SELECT guid FROM item_instance);
```

**Comment by moncat2005** [comment]
I tried executing the SQL command
SELECT * FROM characters_npcbot_gear_storage WHERE item_guid NOT IN (SELECT guid FROM item_instance);
The result returns 0
